### ISSUE TITLE
Compiler bug on `collect`

### ISSUE THREAD

**Issue by yanis-fourel** [nushell/nushell#14680]
### Describe the bug

I get the following "compiler bug" error when I run my script:

```
Error: nu::compile::register_uninitialized

  × Register %0 was uninitialized when used, possibly reused.
   ╭─[/home/yanis/code/aoc2024_nu/day2b.nu:7:28]
 6 │         or (
 7 │             0..($mylist | length) | collect | is-empty
   ·                                     ───┬───
   ·                                        ╰── while adding this instruction: Call { decl_id: DeclId(2), src_dst: RegId(0) }
 8 │         )
   ╰────
  help: this is a compiler bug. Please report it at https://github.com/nushell/nushell/issues/new
        from: crates/nu-engine/src/compile/builder.rs:150:22

Error:   × Can't evaluate block in IR mode
    ╭─[/home/yanis/code/aoc2024_nu/day2b.nu:1:13]
  1 │ ╭─▶ def main [] {
  2 │ │       let mylist = [1 2 3]
  3 │ │
  4 │ │       (
  5 │ │           ($mylist | is-empty)
  6 │ │           or (
  7 │ │               0..($mylist | length) | collect | is-empty
  8 │ │           )
  9 │ │       )
 10 │ ├─▶ }
    · ╰──── block is missing compiled representation
    ╰────
  help: the IrBlock is probably missing due to a compilation error
```

### How to reproduce

script.nu:
```nu
def main [] {
    let mylist = [1 2 3]

    (
        ($mylist | is-empty)
        or (
            0..($mylist | length) | collect | is-empty
        )
    )
}
```
run `nu script.nu`

### Expected behavior

The code should either compile, or if the code is nonsensical then it should say so.  
I'm new to nushell, and I don't yet understand `collect` well.

### Configuration

```
> version | transpose key value | to md --pretty                                                            1
| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.100.0                                                           |
| major              | 0                                                                 |
| minor              | 100                                                               |
| patch              | 0                                                                 |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | linux-x86_64                                                      |
| build_target       | x86_64-unknown-linux-gnu                                          |
| rust_version       | rustc 1.82.0 (f6e511eec 2024-10-15) (built from a source tarball) |
| cargo_version      | cargo 1.82.0 (8f40fc59f 2024-08-21)                               |
| build_time         | 1980-01-01 00:00:00 +00:00                                        |
| build_rust_channel | release                                                           |
| allocator          | mimalloc                                                          |
| features           | default, sqlite, trash                                            |
| installed_plugins  |                                                                   |
```

**Comment by WindSoilder** [comment]
Relative to: https://github.com/nushell/nushell/issues/14675
I think it's already fixed in version 0.101.0
Can you have a check?

**Comment by yanis-fourel** [comment]
I checked out master, manually compiled and tried again. It worked.  
This has already been fixed, closing
